### PR TITLE
bpo-37386: Conditionally skip test_large_file_ops

### PR DESCRIPTION
**Pull request by potomak**
Skip the test if `write()` fails with `errno.ENOSPC`: "OSError: [Errno
28] No space left on device".

https://bugs.python.org/issue37386

### CODE REVIEW

**Review by mangrisano — APPROVED** [review]
LGTM.

File: Lib/test/test_io.py
Comment (dhilst) [inline Lib/test/test_io.py:615]: Do this raise stop the test? Or it just print in the stdout!?

Comment (potomak) [reply Lib/test/test_io.py:615]: `raise` will re-raise the current exception, effectively stopping the single test.

I added a line `raise RuntimeException` in the test body to test this behavior, this is the result:

```
======================================================================
ERROR: test_large_file_ops (test.test_io.CIOTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gcappellotto/src/cpython/Lib/test/test_io.py", line 608, in test_large_file_ops
    raise RuntimeError
RuntimeError

======================================================================
ERROR: test_large_file_ops (test.test_io.PyIOTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gcappellotto/src/cpython/Lib/test/test_io.py", line 608, in test_large_file_ops
    raise RuntimeError
RuntimeError

----------------------------------------------------------------------

Ran 606 tests in 38.472s

FAILED (errors=2, skipped=8)
test test_io failed
test_io failed in 38 sec 661 ms

== Tests result: FAILURE ==

1 test failed:
    test_io

Total duration: 38 sec 714 ms
Tests result: FAILURE
```

**Review by vstinner — APPROVED** [review]
LGTM. But I would prefer to get a second review from another core dev.

**Comment by vstinner** [comment]
@serhiy-storchaka: would you mind to review this change? I think you have some experience with largefile/bigmem tests.

**Comment by vstinner** [comment]
https://bugs.python.org/issue37386 was closed as "not a bug".

**Pull request closed without merging**
